### PR TITLE
GAWB-2877: Test changing billing accounts for billing project owners

### PR DESCRIPTION
**Pull request by MatthewBemis**
This also re-enables the old billing creation test which will hopefully not trigger sign-in challenges.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Google.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.auth.AuthToken
+
+/**
+  * Created by mbemis on 1/5/18.
+  */
+object Google extends FireCloudClient with LazyLogging {
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Google.scala:9]: Unfortunately named `FireCloudClient`. I _think_ we may have renamed this to something more general in our 2801 work? If not, we should make sure to do that.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/billing/BillingSpec.scala [hunk 2]
```diff
@@ -35,6 +38,22 @@ class BillingSpec extends FreeSpec with WebBrowserSpec with UserFixtures with Cl
...
+        //note that this next part of the test *could* be a different test on it's own, but we have chosen to
+        //build it into this test case because:
+        // a) having it in it's own test case would add another ~5 minutes to the test cycle
+        // b) this test would fail anyway if billing project creation failed
+
+        //should you decide to break this out into it's own test, billing project creation for it should
+        //be driven by API and not UI
+
+        val originalBillingAccount = Google.billing.getBillingProjectAccount(billingProjectName)
+        assertResult(Some(Config.Projects.billingAccountId))(originalBillingAccount)
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/billing/BillingSpec.scala]: These tests have been using `should` matchers, i.e.:
```
originalBillingAccount shouldBe Some(Config.Projects.billingAccountId)
```
In other cases, there's some extra power in the `should` matchers that you don't have with `assertResult`, but I guess it doesn't matter in this case.

**Review by breilly2 — APPROVED** [review]
👍

**Review by andy7i — APPROVED** [review]

**Comment by MatthewBemis** [comment]
Jenkins test run here with correct branches: https://fc-jenkins.dsp-techops.broadinstitute.org/job/fiab-test/1831/consoleFull

The 1 test failure is a product of GAWB-2634 (already on dev). Joel is working on fixing this.

**Comment by jmthibault79** [comment]
Here's my swat fix https://github.com/broadinstitute/firecloud-ui/pull/1110

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Google.scala (hunk 1 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/Google.scala:9]: I called it `RestClient` in my attempt from a long time ago https://github.com/broadinstitute/firecloud-ui/pull/986

**Pull request merged**